src/xml_utils.py

```diff
@@ -241,30 +245,42 @@ def generate_new_emails_df(self, dict_data, dev_url):
                        'url', 'authors']
         df_dict = {col: [] for col in (columns + source_cols)}
 
-        for data in range(len(dict_data)):
+        seen_titles = set()
+        for idx in range(len(main_dict_data)):
             xmls_list = []
-            number = get_id(dict_data[data]["_source"]["id"])
-            title = dict_data[data]["_source"]["title"]
-            xml_name = clean_title(title)
-            file_name = f"{number}_{xml_name}.xml"
-            combined_filename = f"combined_{xml_name}.xml"
-
-            if not any(file_name in item for item in files_list):
-                logger.info(f"Not present: {file_name}")
+            this_title = main_dict_data[idx]["_source"]["title"]
+            if this_title in seen_titles:
+                continue
+
+            # generate xml for all the docs that fall under this title
```

Comment:
@urvishp80 I think that this comment is not correct, and it can be misleading for someone that is trying to read the code.
With this new code, you indeed fetch all the documents under this title, but it's not here that you are creating the XML files.